src/core/inductive.ml

```diff
@@ -0,0 +1,65 @@
+(** Generating of induction principles. *)
+
+open Timed
+open Pos
+open Console
+open Terms
+open Print
+
+(** Builtin configuration for induction. *)
+type config =
+  { symb_Prop : sym (** : TYPE. Type of propositions. *)
+  ; symb_prf  : sym (** : Prop → TYPE.
+                        Interpretation of propositions as types. *) }
+
+(** [get_config ss pos] build the configuration using [ss]. *)
+let get_config : Sig_state.t -> Pos.popt -> config = fun ss pos ->
+  let builtin = Builtin.get ss pos in
+  { symb_Prop = builtin "Prop"
+  ; symb_prf  = builtin "P" }
+
+let principle : Sig_state.t -> popt -> sym -> sym list -> term =
+  fun ss pos sind scons_list ->
+  let c = get_config ss pos in
+  let ind = _Symb sind in
+  let prop = _Symb c.symb_Prop in
+  let p = Bindlib.new_var mkfree "p" in
+  let prf_of_p t = _Appl (_Symb c.symb_prf) (_Appl (_Vari p) t) in
+  let app = List.fold_left _Appl in
+
+  let case_of scons =
+    let rec case xs a =
+      match unfold a with
+      | Symb(s) ->
+         if s == sind then prf_of_p (app (_Symb scons) (List.rev xs))
+         else fatal pos "%a is not a constructor of %a"
+                pp_symbol scons pp_symbol sind
+      | Prod(a,b) ->
+         let (x,b) = Bindlib.unbind b in
```

Comment:
At the last meeting, to variable names different from "_", we said that we would perhaps do something like this instead:
```
let (x,b) =
  if Bindlib.binder_occur b then Bindlib.unbind b else
  let x = Bindlib.new_var mkfree "x" in
  (x, Bindlib.subst b (Vari x))
```